split-the-stream/README.md

```diff
@@ -0,0 +1,178 @@
+
+- No extra cost of two deployments and two codebases.
+- New jobs will be implemented as new handlers.
+- We don't have support for multiple identities for one forge.
+  - Not hard to do but requires some work.
+- Fedora-source-git friendly: easy combination of events (different for Fedora and Stream)
+  and handlers (=implementation, can be shared).
+
+### 1. same codebase, new deployment
+
+- No extra cost of maintenance of two codebases.
+- New jobs are implemented as new handlers.
+- Different identities can be used in one git forge (=gitlab.com).
+- Resources can be tweaked separately.
+- Fedora-source-git friendly: easy combination of events (different for Fedora and Stream)
+  and handlers (=implementation, can be shared).
+
+### 2. separate workers
+
+- New jobs are implemented as new handlers in a separate repository.
+- The centos-stream related code is in one place, based-on the packit-service code.
+- We can use same or a separate deployment.
+  (Fedora-source-git can be separated or with Stream.)
+
+### 3. split the packit-service repo and build upstream/centos-stream workers
+
+- One repo with the scheduler.
+  Two repositories with the worker(=handlers) definition: one for the upstream, one for the stream.
+- Requires more work.
+- Can lead to a cleaner architecture. Something we were discussing for some time.
+  - The centos-stream related code is in one place,
+    upstream code is in one place and
+    the shared code is in one place.
+- Another dependency in the chain.
+  - It's sometimes hard to work on the functionality that goes across multiple git projects.
+- We can use same or a separate deployment.
+  (Fedora-source-git can be separated or with Stream.)
+
+### 4. fork and improve
+
+- The benefits of the current service code can be preserved.
+- The non-relevant/bad code can be removed.
+- More time needed for development.
+- Improvements relevant for both are hard to sync.
+- More time needed for maintenance.
+
+### 5. separate project from scratch
+
+- The new service can be more lightweight and efficient.
+- We can iterate on the prototype more quickly.
+- We can get rid of the old bad staff in the packit-service.
+- We can go through the same pain we've already gone through.
+- We need to maintain two separate projects. (We need more people or reduce the productivity.)
+- We are not motivated to improve the current projects.
+- To share the code between the upstream and source project, we need to create some shared libraries.
+  - Can lead to 3. and/or having another project on our dependency chain.
+
+## Linking of the merge-requests
+
+- We can set the dependent merge-request across multiple projects
+  (e.g. https://gitlab.com/lachmanfrantisek/kernel-ark/-/merge_requests/1)
+  - I couldn't find a related API.
+    (The details does not show anything relevant:
+    [example response](https://gitlab.com/api/v4/projects/23847254/merge_requests/1).)
+- Mentioning the other merge-request is always a possibility.
```

Comment:
What about protecting the branch?
(But that also moves the full responsibility for branches to us... just maintainers and packit maybe?)

Reply:
Aha, you mean to not allow the merge?

I think there are two goals with this (I'll update the text to cover that.):
* User can easily find the related merge request on the web UI. (In both ways.)
* We can get the related merge request from the service.